apps/condo/domains/billing/utils/countrySpecificQRCodeParsers/ru.qrCodeParser.js

```diff
@@ -0,0 +1,39 @@
+const iconv = require('iconv-lite')
+const { get } = require('lodash')
+
+/**
+ * @typedef {Object} TRUQRCodeFields
+ * @property {string} BIC
+ * @property {string} PayerAddress
+ * @property {string} PaymPeriod
+ * @property {string} Sum
+ * @property {string} PersAcc The resident's account
+ * @property {string} PayeeINN
+ * @property {string} PersonalAcc The bank account of the receiver
+ */
+
+/**
+ * @param {string} qrStr The QR code string got from the picture
+ * @return {TRUQRCodeFields}
+ */
+function parseRUReceiptQRCode (qrStr) {
+    const matches = /^ST(?<version>\d{4})(?<encodingTag>\d)\|(?<requisitesStr>.*)$/g.exec(qrStr)
+
+    if (!matches) {
+        throw new Error('Invalid QR code')
+    }
+
+    const requisitesStr = get(matches, ['groups', 'requisitesStr'], '')
```

Comment:
What happens if there is no `requisitesStr` in match? How it will be encoded in the future? Don't you need to handle that case and maybe throw an error here?

Reply:
The error will be thrown and catched. As the result user will receive an GQL error (INVALID_QR_CODE_STRING).